pkg/syslogger/syslogger.go

```diff
@@ -0,0 +1,59 @@
+package syslogger
+
+import (
+	"fmt"
+	"log/syslog"
+	"sync"
+)
+
+// interface created for mocking `mock_syslogger.go`
+// used in logger/logger_test.go
+type SyslogWriterInterface interface {
+	Write(p []byte) (n int, err error)
+	Close() error
+}
+
+var _ SyslogWriterInterface = &SyslogWriter{}
+
+type SyslogWriter struct {
+	writer *syslog.Writer
+	mutex  sync.Mutex
+}
+
+func NewSyslogWriter(priority syslog.Priority, tag string) (*SyslogWriter, error) {
+	w, err := syslog.Dial("udp", "0.0.0.0:514", priority, tag)
```

Comment:
We want to use `TCP` so that we don't lose log messages (UDP is fire & forget). If the logger cannot establish a TCP connection, it should panic - [let it crash](https://wiki.c2.com/?LetItCrash). We wan to be sure that the other end is able to receive these logs.

We also want the `HOST:PORT` to come from an environment variable.

The whole purpose of this PR is so that we don't lose log messages. Investing in integration - good thing we have Dagger! - is worth the effort 💪